### PR TITLE
Feature/rest routes

### PR DESCRIPTION
**Pull request by mordrax**
accepts urls like:
http://localhost:8000/Main.elm/#/index/students/1077604
http://localhost:8000/Main.elm/#/index/events

also hacked around button ripple issue where clicking any button rippled all buttons

### CODE REVIEW

File: Main.elm [hunk 1]
```diff
@@ -117,12 +118,46 @@ urlUpdate : String -> Model -> ( Model, Cmd Msg )
...
+            { model | currentPage = page }
+
+        studentsRegex =
+            Regex.regex "index.students.(\\d+)"
+
+        eventsRegex =
+            Regex.regex "index.(events)"
+
+        matches regex =
+            Regex.find Regex.All regex (Debug.log "url" url)
```
Comment (JulianLeviston) [inline Main.elm]: 🔥 debug

File: Pages/Index.elm [hunk 2]
```diff
@@ -188,10 +207,10 @@ type alias ViewButton =
     String -> Msg -> Html Msg
 
 
-btn : String -> Msg -> Material.Model -> Html Msg
-btn txt msg mdl =
+btn : String -> Msg -> Material.Model -> Int -> Html Msg
+btn txt msg mdl parts =
     Button.render MDL
-        [ 0 ]
+        [ parts ]
```
Comment (JulianLeviston) [inline Pages/Index.elm]: Ah :)

**Comment by JulianLeviston** [comment]
🚢 hopefully I'll get a refactor done before we need to add a billion more features :)

File: Main.elm [hunk 3]
```diff
@@ -117,12 +118,46 @@ urlUpdate : String -> Model -> ( Model, Cmd Msg )
...
+                |> List.head
+                |> Maybe.map .submatches
+                |> Maybe.map List.head
     in
-        url
-            |> (\x -> AllDict.get x routes)
-            |> Maybe.withDefault Index
-            |> changeToPage
+        case ( matches studentsRegex, matches eventsRegex ) of
+            ( Just (Just (Just studentIdString)), _ ) ->
```
Comment (mordrax) [inline Main.elm]: HAHAHAHAHAHHAHAA

File: Pages/Index.elm (hunk 2 above)
Comment (mordrax) [inline Pages/Index.elm]: is shit... the only reason why it works on the buttons in the table is that it clears the table before you can see all the buttons ripple

File: Pages/Index.elm [hunk 4]
```diff
@@ -279,10 +298,10 @@ sliderView : History -> Float -> Material.Model -> Html Msg
 sliderView history slider mdl =
     let
         olderHistory =
-            btn "<" (SliderMsg (Basics.max 1 (slider - 1))) mdl
+            btn "<" (SliderMsg (Basics.max 1 (slider - 1))) mdl 2
```
Comment (mordrax) [inline Pages/Index.elm]: yeah but is shit for the reason stated above... seriously.. why? is this a feature?

**Pull request merged**